build-index/tagclient/client.go

```diff
@@ -54,6 +58,21 @@ type singleClient struct {
 	tls  *tls.Config
 }
 
+// Request for ListWithPagination.
+type ListRequest struct {
+	Filter struct {
+		Offset string
+		Limit  int
+	}
+	Query string
+}
+
+// Response from ListWithPagination.
+type ListResponse struct {
```

Comment:
is it possible to remove this struct and just use tagsmodels.ListResponse? Having two similar structs can be confusing.

Reply:
tagclient.ListResposne is internal struct to transfer data between different modules. tagmodels.ListResponse is http response though and contains field that are not relevant to tagclient.ListResponse. Let me rename tagclient.ListResponse to avoid confusion.

Reply:
imo having two similar structs is more confusing then a struct containing irrelevant fields. All you need is `GetNames`, `GetOffset` and `GetNextURL` ?

Reply:
Response from tagserver contains next in url format
e.g "/list/[a-z][A-Z][0-9]*?limt=<n>&offset=<token-string>"
compared tagclient.ListWithPagination that contains offset(read next) as token-string. The reason is to hide it from the caller of tagclient.ListWithPagination. I can consolidate the two struct with Next field being interpreted as 2 different things based on the context?